ktor-http/ktor-http-cio/jvm/src/io/ktor/http/cio/websocket/WebSocketDeflateExtension.kt

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.http.cio.websocket
+
+import io.ktor.http.cio.internals.*
+import io.ktor.util.*
+import java.util.zip.*
+
+private const val SERVER_MAX_WINDOW_BITS: String = "server_max_window_bits"
+private const val CLIENT_NO_CONTEXT_TAKEOVER = "client_no_context_takeover"
+private const val SERVER_NO_CONTEXT_TAKEOVER = "server_no_context_takeover"
+private const val CLIENT_MAX_WINDOW_BITS = "client_max_window_bits"
+private const val PERMESSAGE_DEFLATE = "permessage-deflate"
+
+private const val MAX_WINDOW_BITS: Int = 15
+private const val MIN_WINDOW_BITS: Int = 8
+
+/**
+ * Compress and decompress WebSocket frames to reduce amount of transferred bytes.
+ *
+ * Usage
+ * ```kotlin
+ * install(WebSockets) {
+ *     extensions {
+ *         install(WebSocketDeflateExtension)
+ *     }
+ * }
+ * ```
+ *
+ * Implements WebSocket deflate extension from [RFC-7692](https://tools.ietf.org/html/rfc7692).
+ * This implementation is using window size = 15 due to limitations of [Deflater] implementation.
+ */
+@OptIn(ExperimentalWebSocketExtensionApi::class)
```

Comment:
The extension itself is experimental, right? Or we think that it's safe to use it in spite of that it's based on the experimental API?

Reply:
The extension is safe and using experimental API